src/pages/TokenDetails/index.tsx

```diff
@@ -137,15 +142,17 @@ export default function TokenDetails() {
       })
     : null
 
-  if (token === undefined) {
-    return <Navigate to={{ ...location, pathname: '/tokens' }} replace />
-  }
+  // TODO: Fix this logic to not automatically redirect on refresh, yet still catch invalid addresses
```

Comment:
@JackShort pushed this code & knows about the automatic redirect on refresh issue. I commented it out cause I wanted to save his code but disabled it for now since it's hard to test/demo caching when refresh redirects